### PR TITLE
Subscription for Apps, Filters and Patterns

### PR DESCRIPTION
**Pull request by MUzairS15**
Signed-off-by: MUzairS15 <muzair.shaikh810@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by theBeginner86 — COMMENTED** [review]
While uploading a [sample docker-compose app](https://raw.githubusercontent.com/meshery/meshery/master/docker-compose.yaml), the application page keeps loading and doesn't update the application page with latest available apps. I checked the logs, where I found that new uploading app does gets saved but fetch operations doesn't seem to conclude. After refreshing the page, I do see the uploaded app.
Though, pattern page works fine.

@MUzairS15 did you observed such behaviour?

https://user-images.githubusercontent.com/73700530/180661842-ba35978c-62a3-4762-91cf-76b59e58495e.mov

**Comment by MUzairS15** [comment]
@theBeginner86 This new subscription should not affect Meshery UI as it is meant for MeshMap, though we require for Meshery UI also, but for now it is not (also in Meshery UI there is not active update for the application file)
Hence, the behaviour is not related to change in this PR.

**Comment by theBeginner86** [comment]
> @theBeginner86 This new subscription should not affect Meshery UI as it is meant for MeshMap, though we require for Meshery UI also, but for now it is not (also in Meshery UI there is not active update for the application file) Hence, the behaviour is not related to change in this PR.

I wasn't experiencing this behaviour on master so thought this might have been something that went out of the radar.
Thanks for confirming!

**Review by theBeginner86 — APPROVED** [review]

**Comment by MUzairS15** [comment]
@theBeginner86 Thankyou, I was missing nil check, hence call was getting blocked.

**Review by leecalcote — DISMISSED** [review]

File: internal/graphql/resolver/filter.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/layer5io/meshery/handlers"
+	"github.com/layer5io/meshery/internal/graphql/model"
+	"github.com/layer5io/meshery/models"
+)
+
+func (r *Resolver) fetchFilters(ctx context.Context, provider models.Provider, selector model.PageFilter) (*model.FilterPage, error) {
```
Comment (leecalcote) [inline internal/graphql/resolver/filter.go:12]: @MUzairS15 lint check issue.

**Comment by leecalcote** [comment]
Refreshing in the hopes that tests pass

**Comment by MUzairS15** [comment]
Data Type of applications updated wrt to data type in `source_type` PR

**Review by leecalcote — APPROVED** [review]

**Review by Abhishek-kumar09 — APPROVED** [review]
LGTM, Thanks for the big guy @MUzairS15

File: internal/graphql/resolver/configuration.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	if r.Config.FiltersChannel == nil {
+		r.Config.FiltersChannel = make(chan struct{})
+	}
+	configuration := make(chan *model.ConfigurationPage)
+	go func() {
+		r.Log.Info("Configuration subscription started")
+		for {
+			select {
+			case <-r.Config.PatternsChannel:
+				patterns, err = r.fetchPatterns(ctx, provider, selector)
```
Comment (Abhishek-kumar09) [inline internal/graphql/resolver/configuration.go:34]: this one

**Pull request merged**